lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -4575,19 +4584,34 @@ class DecomposeAtenAdaptiveAvgPool1dOp
     MLIRContext *context = op.getContext();
 
     Value input = op.getSelf();
+
     std::optional<unsigned> maybeRank = getTensorRank(input);
     if (!maybeRank) {
       return rewriter.notifyMatchFailure(op, "expected input to have a rank");
     }
     unsigned rank = *maybeRank;
-    Value sizeDim = rewriter.create<Torch::ConstantIntOp>(
-        loc, rewriter.getI64IntegerAttr(rank - 1));
-    Value inputSize = rewriter.create<AtenSizeIntOp>(loc, input, sizeDim);
 
     Value outputShape = op.getOutputSize();
     SmallVector<Value> outputShapeSizesTorchInt;
     getListConstructElements(outputShape, outputShapeSizesTorchInt);
     Value outputSize = outputShapeSizesTorchInt[0];
+    int64_t outputSizeInt;
```

Comment:
I'm a bit confused by this. Why do we need to change `DecomposeCOmplexOps`? Your changes to the [backend legal ops set](https://github.com/llvm/torch-mlir/pull/2661/files#diff-ed49f70c396f47c0d651d573b089e78db6177242f2d9fd799cefee6212fc6036) should prevent this pattern from running when using `linalg` as the backend.

Reply:
I thought that while running linalg as the backend for the e2e tests, it still tries to apply the -torch-decompose-complex-ops pass. If it runs into the runtime assert op, it causes the test to get stuck in a loop and it doesn't pass the IR to my -torch-to-linalg conversion. Trying to revert the changes to just the DecomposeComplexOps.cpp file and running the e2e tests, the terminal just spits out:

```
unimplemented: only support cases where input and output size are equal for non-unit output size
```

and then sits forever. 

If I use a match failure instead, then it just keeps the unconverted IR, which is legal, and gets passed to my conversion and the tests all pass.

Reply:
Ah!

The reason for my confusion is that I spelled the op wrong in the backend legal ops set, so it basically wasn't doing anything.

Let me make some changes.